site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+This is pretty verbose, but the extra information will be very important once we
+begin having errors with our configuration.
+
+### Multiple Gateways, same GatewayClass, domain match error
+
+In this example, the HTTPRoute `http-parent` is able to attach to either Gateway by itself, but the Gateways share a GatewayClass (and `controllerName`).
+
+However, because both Gateways include different `hostname` fields, once the child is included, `gateway-a` will accept it, and `gateway-b` will not, because the hostnames don't match.
+
+This is a slightly contrived example, but serves to illustrate that we need a way for both `http-parent` and `http-child` to indicate that something has gone wrong.
+
+```yaml
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-a
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: *.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-b
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: subdomain.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-parent
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  - name: gateway-b
+    sectionName: http
+  rules:
+    - allowedRoutes:
+        kinds: 
+          - kind: HTTPRoute
+        namespaces:
+          from: Same
+    - backendRefs:
+      - name: default-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: somevalue
+    parentRef:
+      name: gateway-a
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+  - controllerName: somevalue
+    parentRef:
+      name: gateway-b
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-child
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  hostnames:
+    - testdomain.example.com
+  rules:
+    - backendRefs:
+      - name: foo-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: somevalue
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: GrandparentConfigError
+      message: Some configuration was not accepted in the attachment process. See grandparents for more details.
+    grandparents:
+      - controllerName: somevalue
+        parentRef:
+          name: gateway-a
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          message: Successfully accepted
+      - controllerName: somevalue
+        parentRef:
+          name: gateway-b
+        conditions:
+        - type: Accepted
+          status: false
+          reason: HostnameMismatch
```

Comment:
Listener hostname mismatch doesn't trigger an error so it seems like this shouldn't either?

Reply:
Wouldn't this be parallel/identical to the `NoMatchingListenerHostname` [RouteConditionReason](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.RouteConditionReason)?